stream-listener-json/src/main/java/demo/SampleSink.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package demo;
+
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.Input;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.messaging.SubscribableChannel;
+
+/**
+ * @author Ilayaperumal Gopinathan
+ */
+@EnableBinding(SampleSink.Sink.class)
+public class SampleSink {
+
+	// Sink application definition
+	@StreamListener(Sink.SAMPLE)
+	public void receive(Bar barMessage) {
+		System.out.println("******************");
+		System.out.println("At the Sink");
+		System.out.println("******************");
+		System.out.println("Received transformed message value: " + barMessage.getValue() + " of type " + barMessage.getClass());
+		System.out.println("******************");
+	}
+
+	public interface Sink {
```

Comment:
My intention was to showcase the way someone could create a interface with the custom Channel definitions. But I agree, this is not the scope of this sample and could confuse the user.